gnocchi/storage/__init__.py

```diff
@@ -199,35 +200,55 @@ def _version_check(name, v):
         """
         return name.split("_")[-1] == 'v%s' % v
 
+    def get_aggregated_measures(self, metric, aggregations,
+                                from_timestamp=None, to_timestamp=None):
+        """Get aggregated measures from a metric.
+
+        :param metric: The metric measured.
+        :param aggregations: The aggregations to retrieve.
+        :param from timestamp: The timestamp to get the measure from.
+        :param to timestamp: The timestamp to get the measure to.
+        """
+        timeseries = utils.parallel_map(
+            self._get_measures_timeserie,
+            ((metric, agg, from_timestamp, to_timestamp)
+             for agg in aggregations))
+        return {
+            agg: ts.fetch(from_timestamp, to_timestamp)
+            for agg, ts in six.moves.zip(aggregations, timeseries)
+        }
+
     def get_measures(self, metric, aggregations,
                      from_timestamp=None, to_timestamp=None,
                      resample=None):
-        """Get a measure to a metric.
+        """Get aggregated measures from a metric.
+
+        Deprecated. Use `get_aggregated_measures` instead.
 
         :param metric: The metric measured.
         :param aggregations: The aggregations to retrieve.
         :param from timestamp: The timestamp to get the measure from.
         :param to timestamp: The timestamp to get the measure to.
         :param resample: The granularity to resample to.
         """
-        agg_timeseries = utils.parallel_map(
-            self._get_measures_timeserie,
-            ((metric, ag, from_timestamp, to_timestamp)
-             for ag in aggregations))
+        timeseries = self.get_aggregated_measures(
+            metric, aggregations,
+            from_timestamp=None, to_timestamp=None)
```

Comment:
how come you don't pass in timestamps here?

Reply:
Good catch.